parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -39,17 +87,136 @@ decl_error! {
 decl_module! {
 	pub struct Module<T: Config> for enum Call where origin: T::Origin {
 		type Error = Error<T>;
+
 		fn deposit_event() = default;
+
+		// Generate a message commitment every [`Interval`] blocks.
+		//
+		// The commitment hash is included in an [`AuxiliaryDigestItem`] in the block header,
+		// with the corresponding commitment is persisted offchain.
+		fn on_initialize(now: T::BlockNumber) -> Weight {
+			if (now % Self::interval()).is_zero() {
+				Self::commit_messages()
+			} else {
+				0
+			}
+		}
 	}
 }
 
+pub fn offchain_key(prefix: &[u8], hash: H256) -> Vec<u8> {
+	(prefix, ChannelId::Basic, hash).encode()
+}
+
 impl<T: Config> Module<T> {
-	pub fn submit(_: &T::AccountId, target: H160, payload: &[u8]) -> DispatchResult {
-		Nonce::try_mutate(|nonce| -> DispatchResult {
+	pub fn submit(account_id: &T::AccountId, target: H160, payload: &[u8]) -> DispatchResult {
+		Nonces::<T>::try_mutate(account_id, |nonce| -> DispatchResult {
 			*nonce += 1;
-			T::MessageCommitment::add(ChannelId::Basic, target, *nonce, payload)?;
+			Self::push_message(account_id, target, *nonce, payload)?;
 			<Module<T>>::deposit_event(Event::MessageAccepted(*nonce));
 			Ok(())
 		})
 	}
+
+	fn push_message(account_id: &T::AccountId, target: H160, nonce: u64, payload: &[u8]) -> DispatchResult {
+		let mut mq = MessageQueue::<T>::get();
+		mq.push((
+			account_id.clone(),
+			Message {
+				target,
+				nonce,
+				payload: payload.to_vec(),
+			},
+		));
+		MessageQueue::<T>::put(mq);
+		Ok(())
+	}
+
+	fn commit_messages() -> Weight {
+		let mut all_messages: Vec<(T::AccountId, Message)> =
+			<Self as Store>::MessageQueue::get();
```

Comment:
why the change to `get()` instead of `take()`? With take, the storage is cleared after the read so you don't need to clear it again later in this method.

Reply:
oh, I just didn't know of the existence of take(). I'll change it.

Reply:
It's done now in the newer branch